### DISCUSSION TITLE
What is Arena Model ?

### DISCUSSION THREAD

**Discussion by degletnour** [open-webui/open-webui#6538] [Q&A] [answered]
I couldn't understand the little caption stating that " Anonymous chat bots and vote ". 

What are those anonymous chat bots ? and how do I vote ? and why it is the default option for any new chat ?

Thanks in advance

**Comment by Crear12** [comment]
I'm also wondering about it.......I only have llama3.1 70B on my computer, but my Arena model claims that it's llama2 13B. It looks like it will randomly download and activate one every time the service restarts, but from the memory usage it doesn't make any sense for a 13B model to consume 40GB.

**Reply by rafalpietrzakio** [reply]
Arena model seem to be randomly using from all models you have in your open webUI for each response. You can simply ask "which model are you". Each refresh tends to give different answer. Not really sure about actual value added of such approach

**Comment by ArnaudPolitico** [comment]
Yeah, I think we need more information on this and there is nothing referenced to the arena mod in the documentation.

**Comment by trinhkvo** [comment, marked as answer]
The Area model is an umbrella term for a collection of models designed to compare their responses. It will randomly select a model from the collection to provide an answer to your query. Once you rank the response (use like or dislike button), the Arena model will reveal which specific model generated it, making this process a blind test. You can regenerate the response to obtain another one from a different model for comparison (note that the second response could come from the same model as the first).

If you prefer not to regenerate responses, you can add the same Arena model within the same chat window to receive multiple responses at once.

In Admin Panel > Settings > Evaluations, you can choose to add different models that you have to this Arena model group.

**Reply by Crear12** [reply]
Thanks for answering~
For the "It will randomly select a model from the collection to provide an answer to your query", I'm confused about "the collection" here. Is the collection from online resources, or is it from my local drive (e.g. downloaded llm weights)? 
If it's from my local drive, did it teach the model to hide its identity? My Arena says it's llama2, while the only one that the web interface shows available was llama3.1 70b. I also verified the knowledge cutoff and it only knows about iPhone 13...

**Reply by trinhkvo** [reply]
The Arena model is not teaching anything. Think of it as a "masked" name for model(s) that are available in the Open WebUI instance you are using. They can be the local models you have on your PC, or the online model if you are using API. The admin will select which model(s) to include in that Arena collection in the Admin Panel.
It sounds like you are not the admin, so the model you can select to use and the model included may be different.

**Reply by ballerburg9005** [reply]
It is deceptive that it is called "arena" and not simply "random", because it makes the impression that those models are all externally provided by companies and enthusiasts who want to test and data mine inside an arena, that is naturally competitively scored by the Open-Webui project for the user's benefit. Such an arena feature would be very interesting and useful. The helper hint even states "submit responses to *anonymous* chatbots", implying that you are ultimately not privileged to fully identify them, which again would be only possible if externally provided.

However in actuality, it more or less simply randomizes the model selection and reveals the model's name only after you vote on the response.

**Comment by kernelpanic19** [comment]
In v0.3.35 I am encountering the following when using the Arena model (default Arena model config):

1. The Arena model, after voting, does not reveal the underlying model in the chat nor on the feedback history
2. The Arena model, after voting in a side-by-side chat or after regenerating, does not update the leaderboard (it does update the feedback history).  The leaderboard has only the actual models, and their stats only update when they are voted on as their actual model name (not when they are used by the Arena model).

It's entirely possible I'm doing something wrong, or missing something.  Because at present it's not clear to me how the Arena model can be used to anonymously evaluate the models.

**Comment by tjbck** [comment]
Hope this helps! https://docs.openwebui.com/features/evaluation/

**Reply by ArnaudPolitico** [reply]
> Hope this helps! https://docs.openwebui.com/tutorials/evaluation/

Did you just add it to the doc ? Bc I assume I tried to find out this a few days ago and didn't found anything ? 
Anyway, thank you !

**Reply by kernelpanic19** [reply]
Great doc, thanks! Evaluations is a fantastic feature.

However, I am still confused by the implementation of the Arena Model.  Perhaps I'm misunderstanding, or missing something.
	
- When I do side-by-side rating with an Arena Model and a native model, only the native model is recorded to the leaderboard. There is no corresponding leaderboard win/loss recorded for the Arena Model.
- I would expect the Arena Model win/loss to be recorded in the leader board as its unmasked underlying model. I have not identified a scenario where an Arena Model (masked or unmasked) appears on the leaderboard.
- Is there anywhere that the Arena Model is unmasked, revealing the underlying model to the user?
- If the above is not in place, how does the Arena Model help evaluation?

Also:
- Seems it would be beneficial to allow an Arena Model to compete against itself in sibling responses and be eligible for inclusion on the leaderboard since the masked models should be different. That would allow for a side by side blind comparison. Alternatively one could create two custom Arena Models.

**Reply by GrayXu** [reply]
Currently, anonymous models are not being counted, and models that have switched prompts are also appearing as separate models. It feels like the current arena is still in an unusable beta stage. Note: The accuracy of the Elo ranking relies on each model having a significant number of matches, but this is difficult to achieve in personal deployment scenarios.

**Comment by jepjoo** [comment]
Can I remove Arena model from the list somehow? I too often use it accidentally when I don't want to.

edit: okay found the off switch: Admin panel -> settings -> evaluations

**Reply by df-cgdm** [reply]
The leadeboard score remains empty (no score). Is it normal?
